jina/peapods/runtimes/head/__init__.py

```diff
@@ -235,7 +247,7 @@ async def _handle_data_request(
         elif len(worker_results) > 1:
             DataRequestHandler.replace_docs(
                 response_request,
-                docs=get_docs_from_request(worker_results, field='docs'),
+                docs=docs,
```

Comment:
Lets move the other code here? I mean the reduce logic inside the elif. Maybe even inside the DataRequestHandler?
If it is before the if we may do unnecessary work (ok actually you are checking correctly in your if, but the place is still a bit surprising to me ;))